AEPMessaging/Sources/MessagingMigrator.swift

```diff
@@ -10,46 +10,33 @@
  governing permissions and limitations under the License.
  */
 
-import AEPCore
 import AEPServices
 import Foundation
 
-extension Messaging {
-    /// Loads propositions from persistence into memory then hydrates the messaging rules engine
-    func loadCachedPropositions(for expectedSurface: String) {
+struct MessagingMigrator {
+    static func migrate(cache: Cache) {
         guard let cachedPropositions = cache.get(key: MessagingConstants.Caches.PROPOSITIONS) else {
-            Log.trace(label: MessagingConstants.LOG_TAG, "Unable to load cached messages - cache file not found.")
+            Log.trace(label: MessagingConstants.LOG_TAG, "Unable to load cached messages, cache file not found.")
             return
         }
 
         let decoder = JSONDecoder()
         guard let propositions: [PropositionPayload] = try? decoder.decode([PropositionPayload].self, from: cachedPropositions.data) else {
+            Log.trace(label: MessagingConstants.LOG_TAG, "Unable to decode cached messages.")
             return
         }
 
-        Log.trace(label: MessagingConstants.LOG_TAG, "Loading in-app message definition from cache.")
-        let rules = parsePropositions(propositions, expectedSurfaces: [expectedSurface], clearExisting: false, persistChanges: false)
-        rulesEngine.launchRulesEngine.loadRules(rules, clearExisting: false)
-    }
-
-    func cachePropositions(shouldReset: Bool = false) {
-        // remove cached propositions if shouldReset is true
-        guard !shouldReset else {
-            do {
-                try cache.remove(key: MessagingConstants.Caches.PROPOSITIONS)
-                Log.trace(label: MessagingConstants.LOG_TAG, "In-app messaging cache has been deleted.")
-            } catch let error as NSError {
-                Log.trace(label: MessagingConstants.LOG_TAG, "Unable to remove in-app messaging cache: \(error).")
-            }
-
+        let mappedPropositions = propositions.map { $0.convertToProposition() }
```

Comment:
i'm not following the conversion of `PropositionPayload` to `Proposition` through this pr - they are effectively the same object

Reply:
PropositionPayload -> Proposition

- Better aligned with actual edge response payload
- flattened for ease of use
- struct -> class since we are adding behavior as well where needed